crates/accelerate/src/consolidate_blocks.rs

```diff
@@ -28,12 +25,21 @@ use qiskit_circuit::imports::{QI_OPERATOR, QUANTUM_CIRCUIT};
 use qiskit_circuit::operations::{ArrayType, Operation, Param, UnitaryGate};
 use qiskit_circuit::packed_instruction::PackedOperation;
 use qiskit_circuit::Qubit;
+use rustworkx_core::petgraph::stable_graph::NodeIndex;
+use smallvec::smallvec;
 
 use crate::convert_2q_block_matrix::{blocks_to_matrix, get_matrix_from_inst};
 use crate::euler_one_qubit_decomposer::matmul_1q;
 use crate::nlayout::PhysicalQubit;
 use crate::target_transpiler::Target;
-use crate::two_qubit_decompose::TwoQubitBasisDecomposer;
+use crate::two_qubit_decompose::{TwoQubitBasisDecomposer, TwoQubitControlledUDecomposer};
+
+#[derive(Clone, Debug)]
+#[pyclass]
+pub enum DecomposerType {
```

Comment:
This doesn't actually need to be a pyclass you just need to add `FromPyObject` to the derive line.

```suggestion
#[derive(Clone, Debug, FromPyObject)]
pub enum DecomposerType {
```

That tells pyo3 how to wrap either a `TwoQubitBasisDecomposer` or `TwoQubitControlledUDecomposer` pyclass object the function receives in the enum type.